cpp/memilio/epidemiology/state_age_function.h

```diff
@@ -145,15 +160,87 @@ struct StateAgeFunction {
      * support when the user asks for it, a cached value is used. If m_support_max is set to -1, the cached value is
      * deleted and a recomputation is done the next time the user asks for the support. As the support (potentially)
      * depends on the m_parameter object, the cached value has to be deleted. For details see get_support_max().
+     * The same applies to the m_mean object. See get_mean().
      *
      *@param[in] new_parameter New parameter for StateAgeFunction.
      */
     void set_parameter(ScalarType new_parameter)
     {
-        m_parameter = new_parameter;
+        m_parameter   = new_parameter;
+        m_support_max = -1.;
+        m_support_tol = -1.;
```

Comment:
Is it necessary to set the tolerances to -1 as well? (This also holds for the other setters.)

Reply:
Actually not necessary, no.